ssl/ssl_init.c

```diff
@@ -40,7 +42,7 @@ DEFINE_RUN_ONCE_STATIC(ossl_init_ssl_base)
      * We ignore an error return here. Not much we can do - but not that bad
      * either. We can still safely continue.
      */
-    OPENSSL_atexit(ssl_library_stop);
+    atexit(ssl_library_stop);
```

Comment:
Hmm... this is wrong. We cannot use atexit() here as we would not remove use of it from the library.

Reply:
This was diff in progress I forgot to commit the final change before the push (poor git driving). The interesting fact about `atexit()` is that it is a sufficient workaround to avoid a crash in `shlibloadtest` number 1 on OpenBSD. Using a libc's `atexit()` here allows libc to detect if libssl module is still present. If `OPENSSL_atexit()` is used here the libc can be goofed to call a dead memory when `shlibloadtest` terminates.

I'll push changes with library destructors shortly.